src/nspawn/nspawn.c

```diff
@@ -3426,7 +3426,8 @@ static int outer_child(
         if (r < 0)
                 return r;
 
-        if (arg_read_only && arg_volatile_mode == VOLATILE_NO) {
+        if (arg_read_only && arg_volatile_mode == VOLATILE_NO &&
+                !has_root_mount(arg_custom_mounts, arg_n_custom_mounts)) {
                 r = bind_remount_recursive(directory, MS_RDONLY, MS_RDONLY, NULL);
```

Comment:
i wonder what do do about the check in etc_writable()?

Hmm, if arg_volatile_mode is != VOLATILE_NO we imply arg_read_only btw. Shouldn't we do the same when the root dir is an overlay?

Reply:
That seems intuitive, I'll implement that as well in this PR.

Reply:
Maybe simply check the permissions on /etc? Not sure if there's a helper function for doing that.